packages/story-editor/src/app/rightClickMenu/hooks/useCopyPasteActions.js

```diff
@@ -80,41 +107,42 @@ const useCopyPasteActions = () => {
 
         trackEvent('context_menu_action', {
           name: 'undo_copy_styles',
-          element: selectedElement?.type,
-          isBackground: selectedElement?.isBackground,
+          element: selectedElementType,
+          isBackground: isSelectedElementBackground,
         });
       },
       actionHelpText: UNDO_HELP_TEXT,
     });
 
     trackEvent('context_menu_action', {
       name: 'copy_styles',
-      element: selectedElement?.type,
-      isBackground: selectedElement?.isBackground,
+      element: selectedElementType,
+      isBackground: isSelectedElementBackground,
     });
-  }, [copyElementById, selectedElement, showSnackbar]);
+  }, [
+    copySelectedElement,
+    showSnackbar,
+    selectedElementType,
+    isSelectedElementBackground,
+  ]);
 
-  const selectedElementId = selectedElement?.id;
   const pushUpdate = useCallback(
     (update, commitValues) => {
-      if (selectedElementId) {
-        updateElementsById({
-          elementIds: [selectedElementId],
-          properties: (element) => {
-            const updates = updateProperties(element, update, commitValues);
-            const sizeUpdates = getUpdatedSizeAndPosition({
-              ...element,
-              ...updates,
-            });
-            return {
-              ...updates,
-              ...sizeUpdates,
-            };
-          },
-        });
-      }
+      updateSelectedElements({
```

Comment:
Just curious (since using similar updater in another PR) -- why is `updateSelectedElements` better than `updateElementsById`?

Reply:
yea great question. so if we need to update selected elements these two are logically equivalent:
```js
const { updateElementsById, selectedElements } = useStory(({ state, actions }) => ({
  updateElementsById: actions.updateElementsById,
  selectedElements: state.selectedElements,
}));

const someCallback = useCallback((...args) => {
  updateElementsById({
    elements: selectedElements,
    properties: args
  })
}, [selectedElements, updateElementsById]);
```
vs
```js
const { updateSelectedElements } = useStory(({ actions }) => ({
  updateSelectedElements: actions.updateSelectedElements,
}));

const someCallback = useCallback((...args) => {
  updateSelectedElements({
    properties: args
  })
}, [updateSelectedElements]);
```
the difference is the latter won't force a re-render on any component consuming this hook for the lifetime of the app since all of our actions are stable, the first one will force a re-render every time `selectedElements` updates.

Here specifically, we're not getting any of the benefit because we're still pulling in `selectedElements` into this hook for other logic, but at least it heads toward that direction.

So tldr; we're not getting improvements from this specific instance, but in general, just using `updateSelectedElements` over `updateElementsById` & `selectedElements` is a better practice for performance

Reply:
> Here specifically, we're not getting any of the benefit because we're still pulling in selectedElements into this hook for other logic, but at least it heads toward that direction.

That's why I was wondering why one is better than the other, makes sense now, thanks for the explanation!